packages/core/src/modules/mdoc/MdocApi.ts

```diff
@@ -0,0 +1,75 @@
+import type { MdocCreateOptions, MdocVerifyOptions } from './MdocOptions'
+import type { MdocRecord } from './repository'
+import type { Query, QueryOptions } from '../../storage/StorageService'
+
+import { AgentContext } from '../../agent'
+import { injectable } from '../../plugins'
+
+import { Mdoc } from './Mdoc'
+import { MdocService } from './MdocService'
+
+/**
+ * @public
+ */
+@injectable()
+export class MdocApi {
+  private agentContext: AgentContext
+  private mdocService: MdocService
+
+  public constructor(agentContext: AgentContext, mdocService: MdocService) {
+    this.agentContext = agentContext
+    this.mdocService = mdocService
+  }
+
+  /**
+   * Create a new Mdoc, with a spcific doctype, namespace, and validity info.
+   *
+   * @param options {MdocCreateOptions}
+   * @returns {Promise<Mdoc>}
+   */
+  public async create(options: MdocCreateOptions) {
```

Comment:
does create also sign? I think in SD JWT API we use `sign` so might be nice to stay consistent?